contracts/CometRewards.sol

```diff
@@ -55,29 +59,64 @@ contract CometRewards {
      * @notice Set the reward token for a Comet instance
      * @param comet The protocol instance
      * @param token The reward token address
+     * @param multiplier The multiplier to apply to convert a unit of accrued tracking to a unit of the reward token
      */
-    function setRewardConfig(address comet, address token) external {
+    function setRewardConfigWithMultiplier(address comet, address token, uint256 multiplier) public {
         if (msg.sender != governor) revert NotPermitted(msg.sender);
         if (rewardConfig[comet].token != address(0)) revert AlreadyConfigured(comet);
 
         uint64 accrualScale = CometInterface(comet).baseAccrualScale();
         uint8 tokenDecimals = ERC20(token).decimals();
         uint64 tokenScale = safe64(10 ** tokenDecimals);
+        uint64 rescaleFactor;
         if (accrualScale > tokenScale) {
-            rewardConfig[comet] = RewardConfig({
-                token: token,
-                rescaleFactor: accrualScale / tokenScale,
-                shouldUpscale: false
-            });
+            rescaleFactor = accrualScale / tokenScale;
+            // Flip from downscale to upscale if the multiplier is greater than the rescale factor
+            if (rescaleFactor * FACTOR_SCALE < multiplier) {
+                rescaleFactor = safe64(multiplier / uint256(rescaleFactor) / FACTOR_SCALE);
+                rewardConfig[comet] = RewardConfig({
+                    token: token,
+                    rescaleFactor: rescaleFactor,
+                    shouldUpscale: true
+                });
+            } else {
+                rescaleFactor = safe64(uint256(rescaleFactor) * FACTOR_SCALE / multiplier);
+                rewardConfig[comet] = RewardConfig({
+                    token: token,
+                    rescaleFactor: rescaleFactor,
+                    shouldUpscale: false
+                });
+            }
         } else {
-            rewardConfig[comet] = RewardConfig({
-                token: token,
-                rescaleFactor: tokenScale / accrualScale,
-                shouldUpscale: true
-            });
+            rescaleFactor = tokenScale / accrualScale;
+            // Flip from upscale to downscale if 1 / multiplier is greater than the rescale factor
+            if (FACTOR_SCALE / multiplier > rescaleFactor) {
+                rescaleFactor = safe64(FACTOR_SCALE / multiplier / uint256(rescaleFactor));
+                rewardConfig[comet] = RewardConfig({
+                    token: token,
+                    rescaleFactor: rescaleFactor,
+                    shouldUpscale: false
+                });
+            } else {
+                rescaleFactor = safe64(uint256(rescaleFactor) * multiplier / FACTOR_SCALE);
+                rewardConfig[comet] = RewardConfig({
+                    token: token,
+                    rescaleFactor: rescaleFactor,
+                    shouldUpscale: true
+                });
+            }
```

Comment:
Yeah definitely more complex, but maybe not *so* bad?

Reply:
Probably acceptable given the fact that it maintains backward compatibility.

Reply:
Refactored the code slightly to try to improve readability

Reply:
Due to an edge case with this implementation that I caught while writing scenarios, I decided to re-implement with a simpler solution. See: https://github.com/compound-finance/comet/pull/702#issuecomment-1468736922